constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintLayout.java

```diff
@@ -3676,4 +3676,42 @@ public String getSceneString() {
         mLayoutWidget.getSceneString(ret);
         return ret.toString();
     }
+
+    public interface ValueModifier {
+        boolean update(int width, int height, int id, View view, LayoutParams delta);
+    }
+
+    ArrayList<ValueModifier> modifiers;
+
+    public void addValueModifier(ValueModifier modifier) {
+        if (modifiers == null) {
+            modifiers = new ArrayList<>();
+        }
+        modifiers.add(modifier);
+    }
+
+    void removeValueModifier(ValueModifier modifier) {
```

Comment:
public void remove...

Should we clear up the array somewhere? I'm guessing Views have a lifecyle, but not very familiar with that.

Reply:
If the view goes it is GCed. 
Most designs will allocated it always but since this will be used about 1 in 10000 apps.
I figure minimize the impact and only allocate on first use. Of those that use this most will never remove the modifier. 
I could see some one stupidly adding and removing on status (on touch down) but then you want to minimize the memory churn. 

Overall I just gave up at that point and thought it is not worth it to be too complex in this area.